bundles/catalogue/metadatacatalogue/instance.js

```diff
@@ -377,6 +378,23 @@ Oskari.clazz.define(
                 }
 
                 if (event.getViewState() === 'close') {
+                    me._removeFeaturesFromMap();
+                }
+            },
+            'Search.TabChangedEvent': function (event) {
+                const me = this;
+
+                if (event.getNewTabId() !== this.id) {
+                    me._stopCoverage();
+                    if (me.coverageButton) {
+                        me.coverageButton.val(me.getLocalization('delimitArea'));
+                    }
+                    me.drawCoverage = true;
+
+                    if (me.coverageButton) {
+                        me.coverageButton[0].data = '';
+                    }
```

Comment:
We should do some teardown() function that does this and call it instead of copypasting the code.